src/udev/udev-event.c

```diff
@@ -574,6 +574,9 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
                                          fd == spawn->fd_stdout ? "out" : "err", *q);
         }
 
+        /* On EOF disable the event source to prevent constant EOF reads */
+        if (l == 0)
+                sd_event_source_set_enabled(s, SD_EVENT_OFF);
```

Comment:
Hmm, I think it would be nicer to reverse this: set the event source to SD_EVENT_ONESHOT and then reenable it only if we didn't see EOF anymore. This should be more robust, in particular as there already is a similar bug in the codepath when read() fails, which should also result in the disabling of the event source (see further up)

Reply:
I did wonder about the error case, I will make this change tomorrow as well